private/bufpkg/bufconfig/bufconfig.go

```diff
@@ -143,6 +143,35 @@ func WriteConfigWithDependencyModuleReferences(dependencyModuleReferences ...buf
 	}
 }
 
+// WriteConfigWithBreakingConfig returns a new WriteConfigOption that sets the breaking change
+// config for the module.
+//
+// If this option is used and the version used will be consistent with the rest of the config.
```

Comment:
nit: If we update the behavior to validate, let's update the wording to match it.
```suggestion
// If this option is used then the version used must be consistent with the rest of the config.
```